File: app/src/main/java/com/example/movietrailer/ui/pager/VideoPagerAdapter.java
package com.example.movietrailer.ui.pager;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.movietrailer.ui.main.MainFragment;

import java.util.List;

public class VideoPagerAdapter extends FragmentStateAdapter {
    private static final int MAX_NUM_PAGES = 5;
    private final List<Long> movieIds;

    public VideoPagerAdapter(@NonNull Fragment fragment, List<Long> movieIds) {
        super(fragment);
        this.movieIds = movieIds;
    }

    @Override
    public int getItemCount() {
        return movieIds.size() >= MAX_NUM_PAGES ? MAX_NUM_PAGES : movieIds.size();
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        Long movieId = (position >= 0 && position < movieIds.size()) ? movieIds.get(position) : null;
        if (movieId != null) {
            return MainFragment.Companion.newInstance(movieId);
        } else {
            return MainFragment.Companion.newInstance(0L);
        }
    }
}

File: app/src/main/java/com/example/movietrailer/ui/pager/VideoPagerAdapter.kt
package com.example.movietrailer.ui.pager

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.movietrailer.ui.main.MainFragment

class VideoPagerAdapter(
    fragment: Fragment,
    val movieIds: List<Long>
) : FragmentStateAdapter(fragment) {
    companion object {
        private const val MAX_NUM_PAGES = 5
    }

    override fun getItemCount(): Int {
        return if (movieIds.size >= MAX_NUM_PAGES) {
            MAX_NUM_PAGES
        } else {
            movieIds.size
        }
    }

    override fun createFragment(position: Int): Fragment {
        val movieId = movieIds.getOrNull(position)
        return if (movieId != null) {
            MainFragment.newInstance(movieId = movieId)
        } else {
            MainFragment.newInstance(movieId = 0)
        }
    }
}